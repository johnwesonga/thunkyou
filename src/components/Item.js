import React from 'react';
// Item is a stateless component that receives
// props from the parent component.
const Item = ({items}) =>{
    return (
        <ul>
        {items.map(function(user, index){
            return <li key={user.id}>{user.first_name}<br/>
                        <img src={user.avatar} alt={user.first_name}/>
                  </li>;
          })}
        </ul>
    );
}


export default Item;