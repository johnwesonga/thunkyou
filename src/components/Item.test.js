import React from 'react';
import Item from './Item';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

describe('Item Component' , () => {
    it('should render a profile', () => {
        const tree = toJson(shallow(<Item items={[]} />));
        expect(tree).toMatchSnapshot();
    });

    it('should match its snapshot with profiles', () => {
        const items = [{
            id: '1', 
            first_name: 'john', 
            avatar: 'foo.jpg'}];
        const tree = renderer.create(
          <Item items={items} />
        ).toJSON();      
        expect(tree).toMatchSnapshot();
    });
});