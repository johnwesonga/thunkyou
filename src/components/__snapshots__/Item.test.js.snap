// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should match its snapshot with profiles 1`] = `
<ul>
  <li>
    john
    <br />
    <img
      alt="john"
      src="foo.jpg"
    />
  </li>
</ul>
`;

exports[`Item Component should render a profile 1`] = `<ul />`;
